src/merkle/mmr/accumulator.rs

```diff
@@ -23,16 +23,37 @@ pub struct MmrPeaks {
     ///      elements and the left most has `2**2`.
     ///    - With 12 leaves, the binary is `0b1100`, this case also has 2 peaks, the
     ///      leftmost tree has `2**3=8` elements, and the right most has `2**2=4` elements.
-    pub num_leaves: usize,
+    pub(crate) num_leaves: usize,
 
-    /// All the peaks of every tree in the MMR forest. The peaks are always ordered by number of
+    /// All the peaks of every tree in the [Mmr] forest. The peaks are always ordered by number of
     /// leaves, starting from the peak with most children, to the one with least.
     ///
     /// Invariant: The length of `peaks` must be equal to the number of true bits in `num_leaves`.
-    pub peaks: Vec<RpoDigest>,
+    pub(crate) peaks: Vec<RpoDigest>,
```

Comment:
It is important for this structure that these values have their invariants maintained. Namely the `num_leaves.count_ones() == peaks.len()`.

Because of the above, I removed public access to these attributes. Added validation to the constructor, and introduced accessor methods.

Reply:
We could even consider making them fully private (I think the check in the constructor is pretty light-weight - so, performance is probably not an issue).

Reply:
This was not done because of performance reasons. It was so that I didn't have to touch all the tests. Updated